registry/storage/driver/s3-aws/s3.go

```diff
@@ -423,7 +442,9 @@ func New(params DriverParameters) (*Driver, error) {
 	})
 
 	if params.RegionEndpoint != "" {
-		awsConfig.WithS3ForcePathStyle(true)
+		if !params.VirtualHostedStyle {
+			awsConfig.WithS3ForcePathStyle(true)
+		}
```

Comment:
Here's a question - do we want to allow users to use the path style without setting a custom region endpoint?

I'm inclined to say no, since the current behavior uses virtual hosted bucket names by default and doesn't let you switch without a custom endpoint. Users can work around by directly providing the S3 endpoint.